### DISCUSSION TITLE
How to pass HTTP request headers into a function calling agent?

### DISCUSSION THREAD

**Discussion by brian-carnot** [langchain-ai/langserve#600] [Q&A]
Hi, I have this basic idea and want to check if this is the best approach or if there are any recommendations for examples for me to look at.

So I have this current usage need:

User chats with agent via langserve server. Then, the Langserve server performs function calling, some of which will make HTTP post requests on behalf of the user.

I want these HTTP requests to be as if the user themself is making them, so my idea is to pass the headers into an object that is kept for the duration for a conversation is open. Is this the right approach?

**Comment by eyurtsev** [comment]
Hi @brian-carnot , you can use `per_req_config_modifier` in add_routes or else lower level `APIHandler` to extract information from the raw request.
